tests/isolated/server_tests.py

```diff
@@ -207,6 +206,13 @@ def step_empty_derivables(self):
             _serialize(self.session.state), _serialize(self.client.data)
         )
 
+    def step_json_encoder(self):
+        enc = FiftyOneJSONEncoder
+        oid = "aaaaaaaaaaaaaaaaaaaaaaaa"
+        self.assertEqual(enc.dumps(ObjectId(oid)), '{"$oid": "%s"}' % oid)
+        self.assertEqual(enc.dumps(float("nan")), "NaN")
+        self.assertEqual(enc.dumps(float("inf")), "Infinity")
```

Comment:
What's using the output of this encoder? If it's being used on the client side, `JSON.parse()` does not accept `NaN`/`Infinity`, so I think this would reintroduce the issue that #262 fixed.

However, I haven't gotten the example from #262 to work on develop yet (and haven't found a relevant error either...). Mostly just wondering if this would cause additional frontend issues. If this is only used on the backend, it should be fine.

Reply:
Right. I will fix, I misunderstood things.

Reply:
I've fixed this from an encoding perspective (walked some things back). Supporting these numbers in the App generally is a real spanner in the works, though. Range sliders currently break the App when these numbers are present, for example.

One possible solution is to add a reviver to `JSON.parse` (reserving `"Infinity"`, `"-Inifinity"`, and `"NaN"`) and then convert to the javascript equivalents. Would still have to somehow to support theoretical ranges in the sliders, though,

Just thinking out loud.

Reply:
Yep, I was just about to comment on that. I'm fine with addressing that later, since it doesn't appear to be related to this PR. Thanks for adjusting the JSON encoding!